db/upgrade.php

```diff
@@ -182,6 +115,73 @@ function xmldb_tincanlaunch_upgrade($oldversion) {
     }
 
+    if ($oldversion < 2015112702) { //New version in version.php
+        // Define field tincanactivityid to be added to tincanlaunch
+        $table = new xmldb_table('tincanlaunch');
+        $field = new xmldb_field('tincanmultipleregs', XMLDB_TYPE_INTEGER, '1', null, XMLDB_NOTNULL, null, '1', 'tincanverbid');
+
+        if (!$dbman->field_exists($table, $field)) {
+            $dbman->add_field($table, $field);
+        }
+
+        $table = new xmldb_table('tincanlaunch_lrs');
+        $field = new xmldb_field('useactoremail', XMLDB_TYPE_INTEGER, '1', null, XMLDB_NOTNULL, null, '1');
+
+        if (!$dbman->field_exists($table, $field)) {
+            $dbman->add_field($table, $field);
+        }
+
+        $table->add_field('customacchp', XMLDB_TYPE_CHAR, '255', null, XMLDB_NOTNULL, null, null);
+        if (!$dbman->field_exists($table, $field)) {
+            $dbman->add_field($table, $field);
+        }
+
+        upgrade_mod_savepoint(true, 2015112702, 'tincanlaunch');
+    }
+
+    if ($oldversion < 2016021502) { //New version in version.php
+        // Define field  to be added to table
+        $table = new xmldb_table('tincanlaunch_lrs');
+        $field = new xmldb_field('watershedlogin', XMLDB_TYPE_CHAR, '255', null, XMLDB_NOTNULL, null, null);
+
+        if (!$dbman->field_exists($table, $field)) {
+            $dbman->add_field($table, $field);
+        }
+
+        $field = new xmldb_field('watershedpass', XMLDB_TYPE_CHAR, '255', null, XMLDB_NOTNULL, null, null);
+        if (!$dbman->field_exists($table, $field)) {
+            $dbman->add_field($table, $field);
+        }
+
+        upgrade_mod_savepoint(true, 2016021502, 'tincanlaunch');
+    }
+
+    if ($oldversion < 2016021508) { //New version in version.php
+
+        // Define table tincanlaunch_credentials to be created.
+        $table = new xmldb_table('tincanlaunch_credentials');
+
+        // Adding fields to table tincanlaunch_credentials.
+        $table->add_field('id', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, XMLDB_SEQUENCE, null);
+        $table->add_field('tincanlaunchid', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, null);
+        $table->add_field('credentialid', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, null);
+        $table->add_field('expiry', XMLDB_TYPE_INTEGER, '10', null, XMLDB_NOTNULL, null, null);
+
+        // Adding keys to table tincanlaunch_lrs.
+        $table->add_key('primary', XMLDB_KEY_PRIMARY, array('id'));
+
+        // Adding indexes to table tincanlaunch_lrs.
+        $table->add_index('tincanlaunch_credentialid', XMLDB_INDEX_NOTUNIQUE, array('credentialid'));
+        $table->add_index('tincanlaunch_tincanlaunchid', XMLDB_INDEX_NOTUNIQUE, array('tincanlaunchid'));
+
+        // Conditionally launch create table for tincanlaunch_lrs.
+        if (!$dbman->table_exists($table)) {
+            $dbman->create_table($table);
+        }
+
+        upgrade_mod_savepoint(true, 2016021508, 'tincanlaunch');
```

Comment:
Moodle remembers the date (version) that last changed anything important in the DB structure of each module, as not on each module version there is a change in DB structure. so the $oldversion actually holds that value.

So, you can regularly change the date/version on the version.php file only, and when you have a change in DB structure you also need to use upgrade_mod_savepoint() here too.
 
Here is a short explanation: 
https://docs.moodle.org/dev/Activity_modules#upgrade.php
And a more detailed one:
https://docs.moodle.org/dev/Upgrade_API#Introduction